lib/Parse/ParseStmt.cpp

```diff
@@ -1063,26 +1051,22 @@ static void parseGuardedPattern(Parser &P, GuardedPattern &result,
         P.CodeCompletion->completePostfixExprBeginning(nullptr);
         break;
       }
-      P.consumeToken();
-    } else {
-      result.ThePattern = patternResult.get();
-      status.setHasCodeCompletion();
-      return;
     }
+    auto loc = P.consumeToken(tok::code_complete);
+    result.ThePattern = new (P.Context) AnyPattern(loc);
```

Comment:
Why do we change from `patternResult` to `AnyPattern`?

Reply:
I just inlined `setErrorResult()` here.

Reply:
Previously if the `else` block above was hit we would use `patternResult`, not call that function.  I'm not sure if this matters - does anything care what the pattern is in code-completion?

Reply:
Previously, for 'then' block, `patternResult` was used at [here](https://github.com/apple/swift/pull/19884/files#diff-820a716f8b9e1992c9bd1f0e60a032cdR1105) because it didn't return. We need to fill AST anyway.

Reply:
Okay.